### DISCUSSION TITLE
Is it plan to provide support to CF Conventions?

### DISCUSSION THREAD

**Discussion by BorjaEst** [imk-toolkit/imk-toolkit#64] [Q&A] [answered]
I see this repo aims to collect all methods from IMK scientists in a single place. However, those methods might nor work if the data sources are not standarized. My recomendation, as those sources probably are going to be climatic data, is to support by **core** sources that follow [CF Conventions](https://cfconventions.org/Data/cf-conventions/cf-conventions-1.9/cf-conventions.html). 

In addition, there is a xarray extension [cf_xarray](https://cf-xarray.readthedocs.io/en/latest/examples/introduction.html) to support this (although it still has to improve a lot to my taste). Another alternative would be to use [scitools-iris](https://scitools-iris.readthedocs.io/en/stable/), however the API is completelly different to xarray.

Also, if we go ahead with this, probably should be important to add some CI/CD tests that ensures the methods produce CF Compilance data. For example using [cf-checker](https://github.com/cedadev/cf-checker) or [compliance-checker](https://github.com/ioos/compliance-checker).

**Comment by ucyo** [comment, marked as answer]
Enforcing compliance with the CF conventions is important, but currently out of scope. The compliance itself is provided by the data being processed and not by the entity processing the data. Since the current objective is to create an environment for processing data at IMK we should concentrate on this rather than enforcing compliance.

**Reply by BorjaEst** [reply]
Well then probably IMK needs to have a standardized way to store the data and the scripts should all work in that data. Otherwise, we will end up with:
 - Repeated methods that do the same but for different data sources.
 - No guaranty that a function will work (specially on internal calls). If I need `anomalies` in my script (from other script) how I am sure that one will work?
 - Harder testing, debugging and support

**Reply by ucyo** [reply]
> * Repeated methods that do the same but for different data sources.

I think the danger of this happening is pretty low. We are in control of the scripts being added and if they are doing the same, we can merge them.

> * No guaranty that a function will work (specially on internal calls). If I need `anomalies` in my script (from other script) how I am sure that one will work?

That should not be the case. All functions should be working within the module. Do you mean the cases where some variables are differently called than originally intended? Lets say `temp` instead of `temperature` or `t` or `temp` or `t2m`? This is at the responsibility of the script provider and documented in the docstring which naming convention is assumed.

**Reply by BorjaEst** [reply]
> ets say temp instead of temperature or t or temp or t2m?

yes, exacly that.

> This is at the responsibility of the script provider and documented in the docstring which naming convention is assumed.

True, but on other side, it would be so nice we could use always the same variable names....
When using `iris` or `cf_xarray` you can use the `standard_name` to call the variable (even if the var name is different).

See: https://cf-xarray.readthedocs.io/en/latest/quickstart.html#using-attributes

> ets say temp instead of temperature or t or temp or t2m?

With CF ds.cf['[air_temperature'] would work on all cases!

Let's see how it is going without standardization and if it becomes a mess we know where to go.
